lib/common/graph.js

```diff
@@ -201,9 +201,28 @@ class ConcertoGraphVisitor extends DiagramVisitor {
      * @protected
      */
     visitScalarDeclaration(scalarDeclaration, parameters) {
+        super.visitScalarDeclaration(scalarDeclaration, parameters);
         parameters.graph.addVertex(scalarDeclaration.getFullyQualifiedName());
     }
 
+    /**
+     * Visitor design pattern
+     * @param {Decorator} decorator - the object being visited
+     * @param {Object} parameters  - the parameter
+     * @protected
+     */
+    visitDecorator(decorator, parameters) {
+        const parent = decorator.getParent();
+        const modelFile = parent.getModelFile();
+        decorator.getArguments()?.forEach(argument => {
+            const fqn = modelFile.getFullyQualifiedTypeName(argument.name);
+            if (fqn) {
+                parameters.graph.addVertex(fqn);
+                (parent === modelFile) ? parameters.graph.addEdge(parent.getNamespace(), fqn) : parameters.graph.addEdge(parent.getFullyQualifiedName(), fqn);
```

Comment:
`(parent === modelFile) ? parameters.graph.addEdge(parent.getNamespace(), fqn)`

This allows for a model decorated with a type reference to act as a vertex in a connected graph, causing it to be included when users search for subgraphs. 

This behavior arises because we currently allow models to be decorated with type references, which implicitly suggests a composition. However, since we treat models as packages, we should avoid compositions on the packages themselves.